lib/vanagon/component/dsl.rb

```diff
@@ -424,13 +424,52 @@ def environment(*env) # rubocop:disable Metrics/AbcSize
       def add_preinstall_action(pkg_state, scripts)
         pkg_state = Array(pkg_state)
         scripts = Array(scripts)
-
         if pkg_state.empty? || !(pkg_state - ["install", "upgrade"]).empty?
           raise Vanagon::Error, "#{pkg_state} should be an array containing one or more of ['install', 'upgrade']"
         end
         @component.preinstall_actions << OpenStruct.new(:pkg_state => pkg_state, :scripts => scripts)
       end
 
+      # Adds trigger to run during the install phase of packaging
+      #
+      # @param pkg_state [Array] the state in which the scripts should execute. Can be
+      #   one or multiple of 'install' and 'upgrade'.
+      # @param scripts [Array] the Bourne shell compatible scriptlet(s) to execute
+      # @param pkg [String] the package the trigger will be set in
+      def add_install_triggers(pkg_state, scripts, pkg)
+        pkg_state = Array(pkg_state)
+        scripts = Array(scripts)
+        pkg = String(pkg)
+        if pkg_state.empty? || !(pkg_state - ["install", "upgrade"]).empty?
+          raise Vanagon::Error, "#{pkg_state} should be an array containing one or more of ['install', 'upgrade']"
+        end
+        @component.install_triggers << OpenStruct.new(:pkg_state => pkg_state, :scripts => scripts, :pkg => pkg)
+      end
+
+      # Adds interest trigger based on the specified packaging state and interest name.
+      #
+      # @param pkg_state [String] the trigger specified package state. Can be
+      #   one or multiple of 'install' and 'upgrade'
+      # @param scripts [Array] the scripts to run for the interest trigger
+      # @param interest_name String the name of the interest trigger
+      def add_debian_interest_triggers(pkg_state, scripts, interest_name)
+        pkg_state = Array(pkg_state)
+        scripts = Array(scripts)
+        interest_name = String(interest_name)
+        if pkg_state.empty? || !(pkg_state - ["install", "upgrade"]).empty?
+          raise Vanagon::Error, "#{pkg_state} should be an array containing one or more of ['install', 'upgrade']"
+        end
```

Comment:
Since you do this check in a bunch of places, it may be useful to separate this out into its own function, maybe `valid_pkg_state?`?

Reply:
Ohhh yes I like that

Reply:
Yeah, `#add_install_triggers` and `add_debian_interest_triggers` are like 95% identical; that commonality could probably be plucked out into a private method that they both use, and the remaining operation would be what each method brings to the party. Is that what you were getting at, @mwaggett?

Reply:
I was talking about checking that `pkg_state` contains the appropriate values, since that's used in a handful of these methods.
But yeah, I'm in favor of DRYing things up in all the ways. 😊

Reply:
Some variation of `pkg_state - ["install", "upgrade"]).empty?` is used in trigger definitions a lot throughout this PR. That and the coercion of various inputs to these methods indicates that maybe they could be extracted out into their own data structure (maybe even a module or a class, not conceptually dissimilar to Environment variables) that handles validation. Maybe not for this PR if there's a need for this sooner rather than later, but it's something to think about.